### ISSUE TITLE
`E1101 (no-member)` for wrong class with for partially implemented abstract `classmethod`?

### ISSUE THREAD

**Issue by ReubenFrankel** [pylint-dev/pylint#8908]
### Bug description

Hi, I feel like this a false-positive from Pylint when accessing a property from a class inheriting from an abstract class with a partially implemented abstract `classmethod`.

```py
"""pylint test"""

import abc


class Animal(abc.ABC):
    """Animal class"""

    @property
    def name(self) -> str:
        """Animal name"""

    @classmethod
    @abc.abstractmethod
    def from_dict(cls, animal_dict: dict):
        """Parse an Animal object from a dict"""
        return cls()


class Dog(Animal):
    """Dog class"""

    name = "dog"
    barks = False

    @classmethod
    def from_dict(cls, animal_dict):
        new_dog = super().from_dict(animal_dict)
        new_dog.barks = animal_dict.get("barks")

        return new_dog


dog = Dog.from_dict({"barks": True})

print(dog.name)
print(dog.barks)

```

`python test.py`
```
dog
True
```

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************* Module test
test.py:37:6: E1101: Instance of 'Animal' has no 'barks' member (no-member)

------------------------------------------------------------------
Your code has been rated at 6.67/10 (previous run: 0.00/10, +6.67)
```

### Expected behavior

No `E1101` error.

```

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)

```

### Pylint version

```shell
pylint 2.17.5
astroid 2.15.6
Python 3.7.16 (default, Dec  6 2022, 19:08:09) 
[GCC 11.3.0]
```

### OS / Environment

Ubuntu 22.04
